sdk/iOS/src/MSQueuePushOperation.m

```diff
@@ -225,6 +225,43 @@ - (void) finishTableOperation:(MSTableOperation *)operation item:(NSDictionary *
                                               internalError:storeError];
                 }
             }
+            else {
+                // Operation was successful, but another operation is pending.
+                // To be able to complete the subsequent operation, the version
+                // field needs to be updated to the latest version we just got
+                // from the server.
+                NSError* storeError;
+                
+                // Get the item from the local store, which contains updates that haven't yet been pushed to the server.
+                NSDictionary* storeItem = [self.syncContext.dataSource readTable:operation.tableName
```

Comment:
In the case where the pending operation is a delete, there will be no storeItem.  What you actually want to update is the item in the operation queue.  

You aren't explicitly handling a storeItem = nil case in the below logic.